### ISSUE TITLE
Add support of setScriptSource

### ISSUE THREAD

**Issue by auchenberg** [Microsoft/EdgeDiagnosticsAdapter#5]
Add support of setScriptSource, to enable projects like https://github.com/amokjs/amok to work with Edge.

**Comment by caspervonb** [comment]
Quoting @andysterland from #16 

> The adapter doesn't yet implement them all but they are all on the short term to do list as the platform in Edge has the capabilities (**though setScriptSource only works when not at a breakpoint**).

Does this mean the following scenario will not work?

* Set breakpoint at line 0 of myscript.js
* Rewrite myscript.js with setScriptSource
* Resume execution

Playing around with this approach in Chrome and Node as a way to inject transforms at run-time, it gets tedious having to interface with whatever is the flavor of the month in terms of compiler and/or bundler.

**Comment by andysterland** [comment]
@caspervonb you are correct.

Though with any live editing scenario, at a breakpoint or not, you can always get into a state that is nothing like would happen if the edit was applied and the page reloaded. Specifically changing the code doesn't' change the DOM nor does it reverse any affect the code had. Say, if you had a global function named `foo` and then deleted the code for it the function would still be there and could still be invoked. As you mention closures further complicate the matter :(.

But, you are very right the really useful scenario is when you hit an exception, spot the typo, change it and then set the statement (set statement is awesome!) back one and continue. Alas, it's just not supported in Chakra right now :(.

**Comment by caspervonb** [comment]
> This is somewhat key to the feature being useful, otherwise you'll get unreachable code in that it will have closures you cannot reach and missing captures.

I was referring to an explicit case in my use case, which is basically rewrite scripts as they are parsed to inject instrumentation)

----

So to be clear, Chakra's edit and continue feature, cannot actually do edit and continue? What can it do then?

**Comment by nojvek** [comment]
Chakra has edit on idle. Not edit and continue. They are working on having edit and continue but it requires a large refactor on their side to make it work.

**Comment by caspervonb** [comment]
So basically it takes effect on the next call rather than the current call stack?

**Comment by andysterland** [comment]
@caspervonb exactly. The edit itself won't be carried out till the engine goes back to idle, that is the current stack finishes executing.
